examples/md-flexible/src/ParallelVtkWriter.cpp

```diff
@@ -128,6 +153,93 @@ void ParallelVtkWriter::recordTimestep(const int &currentIteration,
   timestepFile.close();
 }
 
+void ParallelVtkWriter::recordDomainSubdivision(const int &currentIteration,
+                                                const autopas::Configuration &autoPasConfiguration,
+                                                const RegularGridDecomposition &decomposition) {
+  if (_mpiRank == 0) {
+    createPvtsFile(currentIteration, decomposition);
+  }
+
+  std::ostringstream timestepFileName;
+  timestepFileName << _dataFolderPath << _sessionName << "_subdivision_" << _mpiRank << "_" << std::setfill('0')
+                   << std::setw(_maximumNumberOfDigitsInIteration) << currentIteration << ".vts";
+
+  std::ofstream timestepFile;
+  timestepFile.open(timestepFileName.str(), std::ios::out | std::ios::binary);
+
+  if (not timestepFile.is_open()) {
+    throw std::runtime_error("Simulation::writeVTKFile(): Failed to open file \"" + timestepFileName.str() + "\"");
+  }
+
+  const std::array<int, 6> wholeExtent = calculateWholeExtent(decomposition);
+  const std::array<double, 3> localBoxMin = decomposition.getLocalBoxMin();
+  const std::array<double, 3> localBoxMax = decomposition.getLocalBoxMax();
+
+  timestepFile << "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\" ?>\n";
+  timestepFile << "<VTKFile byte_order=\"LittleEndian\" type=\"StructuredGrid\" version=\"0.1\">\n";
+  timestepFile << "  <StructuredGrid WholeExtent=\"" << wholeExtent[0] << " " << wholeExtent[1] << " " << wholeExtent[2]
+               << " " << wholeExtent[3] << " " << wholeExtent[4] << " " << wholeExtent[5] << "\">\n";
+  timestepFile << "    <Piece Extent=\"" << wholeExtent[0] << " " << wholeExtent[1] << " " << wholeExtent[2] << " "
+               << wholeExtent[3] << " " << wholeExtent[4] << " " << wholeExtent[5] << "\">\n";
+  timestepFile << "      <CellData>\n";
+  timestepFile << "        <DataArray type=\"Int32\" Name=\"DomainId\" format=\"ascii\">\n";
+  timestepFile << "          " << decomposition.getDomainIndex() << "\n";
+  timestepFile << "        </DataArray>\n";
+  timestepFile << "        <DataArray type=\"Float32\" Name=\"CellSizeFactor\">\n";
+  timestepFile << "          " << autoPasConfiguration.cellSizeFactor << "\n";
+  timestepFile << "        </DataArray>\n";
+  timestepFile << "        <DataArray type=\"String\" Name=\"Container\">\n";
+  timestepFile << "          " << convertToAsciiString(autoPasConfiguration.container.to_string()) << "\n";
```

Comment:
Is there a specific reason to do this as a lambda function? I think an function in the anonymous namespace would suffice. While lambdas can be great, here it would obscure the code more compared to a normal function. But this sentiment might be subjective. I have not head a "professionel" opinion about it.

Reply:
No there is no specific reason and I don't have a strong opinion. My line of thought why I would do it via a lambda was just
1. The code is so specific that it makes probably no sense to be reused anywhere else so why expose it?
2. The definition is directly where it will be used so looking it up is easier / quicker.
3. If it is a proper function you need a template for the type of data. The lambda can do it via auto. Hence it is less syntax overhead.

Reply:
I will check if the function could be reused. The data should be passed as string, as we need a string in the end anyway. So there is no need for a template. 
If the function cannot be used anywhere else, I will use the lambda.